### ISSUE TITLE
Red line highlight

### ISSUE THREAD

**Issue by eleidan** [AtomLinter/linter-jshint#173]
After using this cool plug-in I found myself irritated with red highlighting. That is, each time I type in, the linter checks code and if the line contains errors, the line is highlighted with red background.
Is there a way to change that behaviour? Frankly, I would like to get rid of it at all.

**Comment by Eschon** [comment]
If you go to the settings of the `linter` package you can disable `Lint on fly`.
When this is disabled it should only display errors when you save the file.

**Comment by steelbrain** [comment]
Closing based on comment above. Thanks @Eschon

**Comment by eleidan** [comment]
@Eschon Unfortunately, it doesn't work.
The `atom-linter` serves well after save, but `linter-jshint` still highlights the line on the fly.
I checked `linter-rubocop`, and it doesn't highlight anything at all.

**Comment by steelbrain** [comment]
@eleidan It's kind-of impossible that `linter` is triggered on save and `linter-jshint` is triggered on the fly :sweat_smile: . That package is actually what asks this package for any errors in the file. Anyways, did you try restarting the editor?
```
mv ~/.atom ~/.atom-backup
apm install linter linter-jshint
```
^ If nothing else works, do that and try to repro.

**Comment by eleidan** [comment]
@steelbrain The `atom-linter` stops checking on the fly and do that **only** after save.
But `linter-jshint` **still** provides checking with error messages in status bar only.
And after save the `linter-jshint` shows error messages in pop up window along with status bar.

**Comment by steelbrain** [comment]
@eleidan Your explanation is confusing. I know some great minds who might know what you're saying. Ping @Arcanemagus @SpainTrain

**Comment by eleidan** [comment]
@steelbrain Well, actually, nothing has changed so far: read my first comment and keep in mind that configuring `atom-linter` does not solve the problem. :wink:

**Comment by SpainTrain** [comment]
I think I am also confused.

@eleidan Could you provide your `config.cson` and some screenshots of what you describe here -



> error messages in status bar only.



> error messages in pop up window

Thanks!

**Comment by Arcanemagus** [comment]
@eleidan By `atom-linter` I'm assuming you mean the `linter` package, as `atom-linter` is a helper library with no UI :wink:

If you disable `Show Inline Tooltips` in the `linter` package settings do these "red highlights" go away? It's possible that whatever theme you are using is causing them to display oddly, this is what they should look like (taken with `Atom Dark`):
![image](https://cloud.githubusercontent.com/assets/427137/10343700/f79d3cac-6cd3-11e5-89bb-a83493183015.png)

As @steelbrain already stated, `linter-jshint` does absolutely nothing on it's own, it merely provides a service that packages like `linter` use to run it. Another possibility is that you have a related package installed that is also running `jshint` on your code? There are quite a few packages that do a similar job to what `linter` + `linter-jshint` do, such as the `jshint` Atom package.

The last possibility that I can think of is that you have a different consumer of the service `linter-jshint` provides installed, such as the `nucleide` package suite.


The way to test the last two hypothesis is to disable `linter-jshint`, restart Atom, and see if you still see these "red highlights".

Let us know what you find!

**Comment by eleidan** [comment]
@Arcanemagus Thank you very much for your kind explanations!

You are right regarding the `linter` package. I mean that very linter from Atom.io community.

That "red highlights" doen not go away If I disable `Show Inline Tooltips` in the `linter` package settings.

And I have no `nucleide` package installed.

And now the problem comes into play:
You are absolutely right, and the `jshint` package is that very suspect that proved to be the source of my problem. I installed it as a requirement for the `linter-jshint`. I thought I *should* do that, you know. And when I started to work, I got the following "red highlights":
![jshint_red_line](https://cloud.githubusercontent.com/assets/540832/10390804/97934b92-6e83-11e5-8224-555a07cea534.png)

The solution is as simple as disabling of the `jshint` package. As I can see, `jshint` does the same job as `linter-jshint`, and I prefer the latter :wink: 

**Summary:** disabling of the `jshint` package solves the problem.

Let me know if you need more information. And thanks again! :+1:

**Comment by Arcanemagus** [comment]
The `jshint` Atom package has nothing to do with `linter`, `linter-jshint` so please close this issue when you get a chance.

As you prefer using `linter-jshint` I would recommend just uninstalling the `jshint` package as it isn't required :wink:

**Comment by steelbrain** [comment]
Closing based on discussion above
